stub-idp/src/main/java/uk/gov/ida/stub/idp/repositories/reaper/StaleSessionReaper.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.ida.stub.idp.repositories.reaper;
+
+import org.apache.log4j.Logger;
+import uk.gov.ida.stub.idp.repositories.IdpSession;
+import uk.gov.ida.stub.idp.repositories.SessionRepository;
+
+import static java.text.MessageFormat.format;
+
+public class StaleSessionReaper implements Runnable {
+
+    private static final Logger LOGGER = Logger.getLogger(StaleSessionReaper.class);
+
+    /**
+     * this operates on all sessions in the database, both eidas and verify
+     */
+    private final SessionRepository<IdpSession> verifySessionRepository;
+    private final StaleSessionReaperConfiguration staleSessionReaperConfiguration;
+
+    public StaleSessionReaper(SessionRepository<IdpSession> verifySessionRepository,
+                              StaleSessionReaperConfiguration staleSessionReaperConfiguration) {
+        this.verifySessionRepository = verifySessionRepository;
+        this.staleSessionReaperConfiguration = staleSessionReaperConfiguration;
+    }
+
+    @Override
+    public void run() {
+        final long sessionsInDatabaseBefore = verifySessionRepository.countSessionsInDatabase();
+        LOGGER.info(format("{0} active sessions before reaping (eidas + verify)", sessionsInDatabaseBefore));
+        final long staleSessionsToReap = verifySessionRepository.countSessionsOlderThan(staleSessionReaperConfiguration.getSessionIsStaleAfter());
```

Comment:
Some of these config parameteres should probably be extracted into class variables or methods to make the code easier to follow.